File: dotnet/Allmark/Render/LinkRenderer.cs
namespace Allmark.Render;

using Allmark.Types;

public static class LinkRenderer
{
    public static OutputRenderer Create()
    {
        return new OutputRenderer
        {
            Name = "link",
            Render = Render,
        };
    }

    public static void Render(MarkdownNode node, RendererState state, bool? decode = true)
    {
        RenderUtils.StartNewLine(node, state);
        var title = !string.IsNullOrEmpty(node.Title) ? $" title=\"{node.Title}\"" : "";
        state.Output.Append($"<a href=\"{Utils.EscapeHtml(node.Info ?? "")}\"{title}>");
        RenderChildren.Execute(node, state);
        state.Output.Append("</a>");
        RenderUtils.EndNewLine(node, state);
    }
}

File: web/src/render-console/linkRenderer.ts
import type MarkdownNode from "../types/MarkdownNode";
import type Renderer from "../types/Renderer";
import type RendererState from "../types/RendererState";
import ANSI from "./ansi";
import renderChildren from "./renderChildren";

const renderer: Renderer = {
	name: "link",
	render,
};
export default renderer;

function render(node: MarkdownNode, state: RendererState): void {
	const style = ANSI.underline + ANSI.blue;
	const reset = ANSI.reset;
	state.output += style;
	renderChildren(node, state);
	if (node.info) {
		state.output += `${reset} ${ANSI.dim}(${node.info})${reset}`;
	} else {
		state.output += reset;
	}
}
